review-problems-sol.ml

```diff
@@ -29,3 +29,43 @@ let rec fold_left f acc l = match l with
 
 let rec rev l = 
   fold_left (fun acc x-> Cons(x,acc)) Nil l
+
+let rec append l1 l2 = match l1 with
+| Nil -> l2
+| Cons(x,xs) -> Cons(x, (append xs l2))
+
+let rec fold_right f l acc = match l with
+  | Nil -> acc
+  | Cons(x,xs) -> f x (fold_right f xs acc)
+
+let fold_left' f e l = 
+	fold_right (fun a b -> f b a) (fold_right (fun a b -> append b (Cons(a, Nil))) l Nil)

```

Comment:
You should only need one call to `fold_right` to pull this off!
Hint: you should use the call to `fold_right` to construct a function that transforms/propagates the accumulator towards the end of the list. This will emulate `fold_left`.

Reply:
Thanks for the feedback! I dropped `fold_left'`. Looking forward to the correct solution:)